theano/tensor/blas.py

```diff
@@ -2149,14 +2149,8 @@ def c_code(self, node, name, inp, out, sub):
         fail = sub["fail"]
 
         # generate contiguity condition
-        def contiguous(var, ndim):
-            strides = "PyArray_STRIDES(%s)" % var
-            return " && ".join([
-                " && ".join("{strides}[{i}] > 0 && {strides}[{i}] % type_size == 0"
-                            .format(strides=strides, i=i) for i in range(ndim)),
-                "(%s)" % " || ".join("{strides}[{i}] == type_size"
-                                     .format(strides=strides, i=i) for i in range(ndim)),
-            ])
+        def contiguous(var):
+            return "(PyArray_IS_C_CONTIGUOUS(%s) || PyArray_IS_F_CONTIGUOUS(%s))" % (var, var)
```

Comment:
This is too string. I think you could just change the range(ndim) to range(1, ndim) in the previous code.